spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageChannel.java

```diff
@@ -85,13 +92,17 @@ protected boolean sendInternal(Message<?> message, long timeout) {
 		catch (TimeoutException e) {
 			return false;
 		}
+		catch (JsonProcessingException e) {
+			logger.error(e.getMessage());
+			throw new MessageDeliveryException(message, e.getMessage(), e.getCause());
+		}
 
 		return true;
 	}
 
-	private SendMessageRequest prepareSendMessageRequest(Message<?> message) {
+	private SendMessageRequest prepareSendMessageRequest(Message<?> message) throws JsonProcessingException {
 		SendMessageRequest sendMessageRequest = new SendMessageRequest(this.queueUrl,
-				String.valueOf(message.getPayload()));
+				message.getPayload() instanceof String ? String.valueOf(message.getPayload()) : objectMapper.writeValueAsString(message.getPayload()));
```

Comment:
I am not sure about having this condition. I would prefer to verify if the `contenttype` is `application/json` instead

Reply:
agree, I have changed condition to verify for `contenttype` instead.